model/src/main/scala/ch.epfl.scala.index.model/SemanticVersion.scala

```diff
@@ -24,20 +24,21 @@ case class OtherPreRelease(o: String) extends PreRelease {
  */
 case class SemanticVersion(
     major: Long,
-    minor: Long = 0,
+    minor: Option[Long] = None,
     patch: Option[Long] = None,
     patch2: Option[Long] = None,
     preRelease: Option[PreRelease] = None,
     metadata: Option[String] = None
 ) extends Ordered[SemanticVersion] {
 
-  def isSemantic = {
+  def isSemantic: Boolean = {
     patch.isDefined &&
-    !patch2.isDefined &&
-    preRelease.map(_.isSemantic).getOrElse(true)
+    patch2.isEmpty &&
+    preRelease.forall(_.isSemantic)
   }
 
   override def toString: String = {
+    val minorPart = minor.map(m => s".$m").getOrElse("")
```

Comment:
```suggestion
    val minorPart = minor.map("." + _).getOrElse("")
```
That would preserve the symmetry with `patchPart` and `patch2Part`.

Alternatively, also amend `patchPart` and `patch2Part` to use a string interpolation like you're doing here.

Reply:
Ok